SC reorg & expand: data, tooling, config (#692)
multiversx/mx-docs: docs/developers/meta/sc-meta-cli.md

- `--no-abi-git-version
    - Skips loading the Git version into the ABI
- `--target-dir-meta`
    - For the meta crates, allows specifying the target directory where the Rust compiler will build the intermediary files. Sharing the same target directory can speed up building multiple contract crates at once.
    - _default_: uses the workspace, or builds in `meta/target`.
- `--target-dir-all`
    - Overrides both the `--target-dir-meta` and the `--target-dir-wasm` args.

[comment]: # (mx-context-auto)

### Calling `info`

The info command prints an overview of the contracts and libraries and residing under a folder. It also prints their framework versions.

As an example, below is the output of calling it in the example contract folder in the framework:

![sc-meta info screenshot](/developers/sc-meta/sc-meta-info.png "Result of calling sc-meta info in the example contract folder in the framework")

Paramameters:
- `--path`
    - Target directory where to call all contract meta crates.
    - _default_: current directory.
- `--ignore`
    - Ignore all directories with these names.
    - _default_: `target`.

[comment]: # (mx-context-auto)

### Calling `upgrade`

> Calling `sc-meta upgrade` will try to automatically upgrade a contract or group of contracts to the latest version.

Reviewer: ```suggestion
Calling `sc-meta upgrade` will try to automatically alter the code of a contract or group of contracts to make it/them compatible with the latest rust framework version.
```

> The oldest version currently supported is `0.28.0`. Any older than that, and the developer will need to upgrade it by hand to `0.28.0`.

Reviewer: ```suggestion
The oldest version currently supported is `0.28.0`. Any older than that, and the developer will need to manually upgrade it to `0.28.0`.
```

It is especially important when upgrading from `0.38` to `0.39.0`, since a lot of changes happened at that point.

:::tip
For projects with multiple contract crates, we recommend upgrading all of them at once. The upgrade algorithm goes step by step, version after version. For some of the major versions, it also checks that the project compiles before moving on. This is to give developers the chance to fix issues manually, if necessary, and not have those issues pile up. If there are local depdencies between contracts, the upgrader will not be able to do the check unless all of them are upgraded together.
:::

> 

Reviewer: ```suggestion

:::caution
Generally, we strongly recommend to ensure code versioning or at least a backup of the contract code to avoid the impossibility of reverting permanent changes. This automatic code altering process involved in using `sc-meta upgrade` highly raises this recommendation.
:::

```

    "commonDependencyPath": "/home/user/multiversx/mx-exchange-sc",
    "dependencies": [
        {
            "path": "common/common_errors",
            "depth": 1
        },
        {
            "path": "common/common_structs",
            "depth": 1
        },
        {
            "path": "common/modules/legacy_token_decode_module",
            "depth": 3
        },
        {
            "path": "common/modules/locking_module",
            "depth": 2
        },
        {
            "path": "common/modules/math",
            "depth": 2
        }
    ]
}
```

Paramameters:
- `--path`
    - Target directory where to call all contract meta crates.
    - _default_: current directory.
- `--ignore`
    - Ignore all directories with these names.
    - _default_: `target`.

[comment]: # (mx-context-auto)

### Calling `new`

<!-- TODO: expand section and move to a separate page -->

Creates a new smart contract project from a standard template.

The tool will replace all necessary names in the project, based on the the project name given by the user. These include:
- the crate name,
- the contract trait name,
- the file name of the main source file.

Paramameters:
- `--template`

>     - The contract template to clone.

Reviewer: ```suggestion
    - The contract template to clone. Available options can be retrieve by using [this](/developers/meta/sc-meta-cli#calling-templates)
```
